barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -314,157 +314,135 @@ std::vector<Row> Execution::gen_trace(std::vector<Instruction> const& instructio
+                                  std::get<uint16_t>(inst.operands.at(3)));
             break;
         case OpCode::EQ_8:
             trace_builder.op_eq(std::get<uint8_t>(inst.operands.at(0)),
+                                std::get<uint8_t>(inst.operands.at(1)),
                                 std::get<uint8_t>(inst.operands.at(2)),
-                                std::get<uint8_t>(inst.operands.at(3)),
-                                std::get<uint8_t>(inst.operands.at(4)),
-                                std::get<AvmMemoryTag>(inst.operands.at(1)));
+                                std::get<uint8_t>(inst.operands.at(3)));
             break;
         case OpCode::EQ_16:
             trace_builder.op_eq(std::get<uint8_t>(inst.operands.at(0)),
+                                std::get<uint16_t>(inst.operands.at(1)),
                                 std::get<uint16_t>(inst.operands.at(2)),
-                                std::get<uint16_t>(inst.operands.at(3)),
-                                std::get<uint16_t>(inst.operands.at(4)),
-                                std::get<AvmMemoryTag>(inst.operands.at(1)));
+                                std::get<uint16_t>(inst.operands.at(3)));
             break;
         case OpCode::LT_8:
             trace_builder.op_lt(std::get<uint8_t>(inst.operands.at(0)),
+                                std::get<uint8_t>(inst.operands.at(1)),
                                 std::get<uint8_t>(inst.operands.at(2)),
-                                std::get<uint8_t>(inst.operands.at(3)),
-                                std::get<uint8_t>(inst.operands.at(4)),
-                                std::get<AvmMemoryTag>(inst.operands.at(1)));
+                                std::get<uint8_t>(inst.operands.at(3)));
             break;
         case OpCode::LT_16:
             trace_builder.op_lt(std::get<uint8_t>(inst.operands.at(0)),
+                                std::get<uint16_t>(inst.operands.at(1)),
                                 std::get<uint16_t>(inst.operands.at(2)),
-                                std::get<uint16_t>(inst.operands.at(3)),
-                                std::get<uint16_t>(inst.operands.at(4)),
-                                std::get<AvmMemoryTag>(inst.operands.at(1)));
+                                std::get<uint16_t>(inst.operands.at(3)));
             break;
         case OpCode::LTE_8:
             trace_builder.op_lte(std::get<uint8_t>(inst.operands.at(0)),
                                  std::get<uint8_t>(inst.operands.at(2)),
-                                 std::get<uint8_t>(inst.operands.at(3)),
-                                 std::get<uint8_t>(inst.operands.at(4)),
-                                 std::get<AvmMemoryTag>(inst.operands.at(1)));
+                                 std::get<uint8_t>(inst.operands.at(2)),
+                                 std::get<uint8_t>(inst.operands.at(3)));
             break;
```

Comment:
Why does lt (8 and 16) have `operands.at(1)` injected after 0 here, but lte 8 does not? Lte 16 also has that.

Reply:
Thanks! That was a bug. Doing massive search/replace has its dangers, and these variants seem to be untested.